src/foam/nanos/auth/SystemAuthService.js

```diff
@@ -11,17 +11,35 @@ foam.CLASS({
 
   javaImports: [
     'foam.nanos.auth.Group',
+    'foam.nanos.session.Session',
     'javax.security.auth.AuthPermission'
   ],
 
+  constants: [
+    { name: 'CACHE_KEY', value: 'system.auth.cache' }
+  ],
+
   methods: [
     {
       name: 'check',
       javaCode: `
         User user = ((Subject) x.get("subject")).getUser();
         Group group = (Group) x.get("group");
-        boolean isSystem = user != null && user.getId() == foam.nanos.auth.User.SYSTEM_USER_ID;
-        return isSystem || group != null && group.implies(x, new AuthPermission("*")) || getDelegate().check(x, permission);
+        if ( user != null && user.getId() == foam.nanos.auth.User.SYSTEM_USER_ID ) {
+          return true;
+        }
+        // Response from group implies is cached to maintain intended performance.
+        if ( group != null ) {
+          Session session = x.get(Session.class);
+          Boolean isSuper = (Boolean) session.getContext().get(CACHE_KEY);
```

Comment:
boolean instead of Boolean

Reply:
Checking for null below - we want Booleans object/reference behaviour unless you have another idea